hack/update-deps.sh

```diff
@@ -7,6 +7,56 @@ set -o pipefail
 cd "$(dirname "${BASH_SOURCE}")/.."
 source "$(dirname "${BASH_SOURCE}")/lib/init.sh"
 
+if [[ -z "${KUBE_BRANCH:-}" ]]; then
+  echo "KUBE_BRANCH must be set to the branch of openshift/kubernetes to build staging repos from (e.g. release-1.8.5)"
+  exit 1
+fi
+
+if [[ -z "${KUBE_TAG:-}" ]]; then
+  echo "KUBE_TAG must be set to the tag of kubernetes/* staging repos to use as a base (e.g. kubernetes-1.8.5)"
+  exit 1
+fi
+
+# Set up mirrors for kubernetes staging repos with carries from openshift/kubernetes
+mirror=${TMPDIR:-/tmp}/openshift-glide-staging-repos
+mkdir -p $mirror || true
+pushd $mirror
+  # Checkout openshift/kubernetes
+  git clone https://github.com/openshift/kubernetes.git kubernetes || true
+  pushd kubernetes
+    git fetch origin --tags
+    git reset --hard
+    git clean -fdx
+    git checkout master
+    git branch -D "${KUBE_BRANCH}" || true
+    git checkout -b "${KUBE_BRANCH}" "origin/${KUBE_BRANCH}"
+  popd
+
+  # Make a repo for each staging dir
+  for pkg in kubernetes/staging/src/k8s.io/*; do
+    dir=$(basename $pkg)
+    echo === $dir
+    # Check out the corresponding kubernetes/* repo
+    git clone https://github.com/kubernetes/$dir.git $dir || true
+    glide mirror set git@github.com:openshift/kubernetes-$dir file://$mirror/$dir --vcs git
```

Comment:
does this permanently dirty my glide state?

Reply:
The mirrors are stored in a mirrors.yaml file in your GLIDE_HOME.

Reply:
> The mirrors are stored in a mirrors.yaml file in your GLIDE_HOME.

That sounds like it permanently dirties my glide state.  Globally even, so its not contained to this repo.